lib/shadow/npm-injection.cjs

```diff
@@ -370,64 +425,70 @@ async function packagesHaveRiskyIssues (registry, pkgs, ora = null, input, outpu
+      const rules = orgSettings.length
+        ? orgSettings[0].issueRules
+        : (await apiKeySettingsPromise).defaultIssueRules
+
       for await (const pkgData of batchScan(pkgs.map(pkg => pkg.pkgid))) {
         let failures = []
+        let warns = []
         if (pkgData.type === 'missing') {
           failures.push({
             type: 'missingDependency'
           })
           continue
         }
         for (const issue of (pkgData.value?.issues ?? [])) {
-          if ([
-            'shellScriptOverride',
-            'gitDependency',
-            'httpDependency',
-            'installScripts',
-            'malware',
-            'didYouMean',
-            'hasNativeCode',
-            'troll',
-            'telemetry',
-            'invalidPackageJSON',
-            'unresolvedRequire',
-          ].includes(issue.type)) {
-            failures.push(issue)
+          if (rules[issue.type]) {
+            if (typeof rules[issue.type] == 'boolean' || rules[issue.type].action === 'error') {
+              failures.push(issue)
+            } else if (rules[issue.type].action == 'warn') {
+              warns.push(issue)
+            }
           }
         }
         // before we ask about problematic issues, check to see if they already existed in the old version
         // if they did, be quiet
-        if (failures.length) {
+        if (failures.length || warns.length) {
           const pkg = pkgs.find(pkg => pkg.pkgid === `${pkgData.pkg}@${pkgData.ver}` && pkg.existing?.startsWith(pkgData.pkg))
           if (pkg?.existing) {
             for await (const oldPkgData of batchScan([pkg.existing])) {
               if (oldPkgData.type === 'success') {
-                failures = failures.filter(
-                  issue => oldPkgData.value.issues.find(oldIssue => oldIssue.type === issue.type) == null
-                )
+                const issueFilter = issue => oldPkgData.value.issues.find(oldIssue => oldIssue.type === issue.type) == null
+                failures = failures.filter(issueFilter)
+                warns = warns.filter(issueFilter)
               }
             }
           }
         }
-        if (failures.length) {
-          failed = true
+        if (failures.length || warns.length) {
+          failed = failures.length > 0
           spinner?.stop()
           translations ??= JSON.parse(fs.readFileSync(path.join(__dirname, '/translations.json'), 'utf-8'))
           formatter ??= new ((await chalkMarkdownPromise).ChalkOrMarkdown)(false)
           const name = pkgData.pkg
           const version = pkgData.ver
-          output.write(`(socket) ${formatter.hyperlink(`${name}@${version}`, `https://socket.dev/npm/package/${name}/overview/${version}`)} contains risks:\n`)
+          output.write(`(socket) ${formatter.hyperlink(`${name}@${version}`, `https://socket.dev/npm/package/${name}/overview/${version}`)} contains ${failures.length ? 'serious ' : ''}risks:\n`)
```

Comment:
this shouldn't give a qualifier as some may be marked as low. if qualifiers are desired we could give the issue levels with each issue